### PR TITLE
Add class skipping based on regex

### LINKED ISSUE #6: Add class skipping based on regex
Add class skipping based on regex

This can help to eliminate unwanted, internal, anonymous classes

### PR DESCRIPTION
**Pull request by kwart**
Resolves #6.

This PR comes with 2 commits:
* first updates Japit to use Java 8 
* the second adds possibility to skip classes which fits a regular expression

Sample:
```bash
# skip classes from internal packages (includes .internal. in name) and anonymous classes (e.g. Type$1)
java -jar japit.jar -cmd COMPARE -r .*\\.internal\\..* -r .*\\\$[0-9]+ -h /tmp/japit-html awesome-1.0.{0,1}.jar
```

### CODE REVIEW

File: src/main/java/org/jboss/japit/analyser/JarAnalyser.java [hunk 1]
```diff
@@ -30,95 +30,101 @@
...
-                                System.err.println("EntryStream couldn't be closed for " + entryName);
-                                ex.printStackTrace(System.err);
+                            ClassPool classPool = new ClassPool(true);
+                            CtClass ctClz = classPool.makeClass(entryStream);
+                            String className = ctClz.getName();
+                            if (selectedFQCN == null || className.equals(selectedFQCN)) {
+
+                                for (Pattern skipPattern : skipPatterns) {
+                                    if (skipPattern.matcher(className).matches()) {
+                                        continue JAR_ELEMENT;
```
Comment (rsvoboda) [inline src/main/java/org/jboss/japit/analyser/JarAnalyser.java]: Labeled continue statement is not nice

I'd rather see new `processEntry` method (using code from line 73 - 117) returning Optional<ClassDetails>
classes.add would be invoked after Optional.ifPresent check

**Review by rsvoboda — CHANGES_REQUESTED** [review]

**Comment by rsvoboda** [comment]
And also would be nice if you could update README with the new option.

And thank you for contribution ;)

File: src/main/java/org/jboss/japit/analyser/JarAnalyser.java (hunk 1 above)
Comment (kwart) [reply src/main/java/org/jboss/japit/analyser/JarAnalyser.java]: IMO the labeled `continue` is much nicer and safer than a simple one. Nevertheless, I can do more Java 8 stuff here ;)

Comment (rsvoboda) [reply src/main/java/org/jboss/japit/analyser/JarAnalyser.java]: Thanks, will check it later today or tomorrow.

I see you have fun with hazelcast/* repositories, enjoy the new challenges!

**Comment by rsvoboda** [comment]
Code reviewed, no performance regression noticed when analyzing test jars

**Pull request merged**